cmd/olympus/actions/merge_db.go

```diff
@@ -0,0 +1,74 @@
+package actions
+
+import (
+	"github.com/gomods/athens/pkg/cdn"
+	"github.com/gomods/athens/pkg/eventlog"
+)
+
+// mergeDB merges diff into the module database.
+//
+// TODO: this is racey if multiple processes are running mergeDB (they will be!) in a few ways:
+//
+// 1. CDN updates that race to change the /list endpoint
+// 2. races between CDN updates and module metadata updates. For example:
+//		- Delete operation deletes from the CDN
+//		- Add operation adds to the CDN and saves to the module metadata DB
+//		- Delete operation adds tombstone to module metadata k/v store
+//
+// Both could be fixed by putting each 'for' loop into a (global) critical section
+func mergeDB(originURL string, diff dbDiff, getter cdn.Getter, saver cdn.Saver, deleter cdn.Deleter) error {
+	for _, added := range diff.Added {
+		if _, err := getter.GetForVersion(added.Module, added.Version); err != nil {
+			// the module/version already exists, is deprecated, or is
+			// tombstoned, so nothing to do
+			break
```

Comment:
you will break here, but what about rest of the collection?

Reply:
good catch - this should be `continue`